scalardb-sample/README.md

```diff
@@ -1,162 +1,126 @@
-      "customer_id": "INT",
-      "timestamp": "BIGINT"
-    }
-  },
-  "sample.statements": {
-    "transaction": true,
-    "partition-key": [
-      "order_id"
-    ],
-    "clustering-key": [
-      "item_id"
-    ],
-    "columns": {
-      "order_id": "TEXT",
-      "item_id": "INT",
-      "count": "INT"
-    }
-  },
-  "sample.items": {
-    "transaction": true,
-    "partition-key": [
-      "item_id"
-    ],
-    "columns": {
-      "item_id": "INT",
-      "name": "TEXT",
-      "price": "INT"
-    }
-  }
-}
-```
+The sample e-commerce application shows how users can order and pay for items by using a line of credit.
 
-All the tables are created in the `sample` namespace.
+The database that you will be using in the sample application is Cassandra. Although Cassandra does not provide ACID transaction capabilities, you can make transactions ACID compliant by having your application connect to the database through ScalarDB.
 
-- `sample.customers`: a table that manages customers' information
-    - `credit_limit`: the maximum amount of money a lender will allow each customer to spend when using a credit card
-    - `credit_total`: the amount of money that each customer has already spent by using the credit card
-- `sample.orders`: a table that manages order information
-- `sample.statements`: a table that manages order statement information
-- `sample.items`: a table that manages information of items to be ordered
+To reference the sample application source code, see [src/main/java/sample/Sample.java](src/main/java/sample/Sample.java).
 
-The Entity Relationship Diagram for the schema is as follows:
+{% capture notice--info %}
+**Note**
 
-![ERD](images/ERD.png)
+Since the focus of the sample application is to demonstrate using ScalarDB, application-specific error handling, authentication processing, and similar functions are not included in the sample application. For details about exception handling in ScalarDB, see [Handle exceptions](https://github.com/scalar-labs/scalardb/blob/master/docs/api-guide.md#handle-exceptions).
+{% endcapture %}
+
+<div class="notice--info">{{ notice--info | markdownify }}</div>
+
+### What you can do in this sample application
+
+The sample application supports the following types of transactions:
+
+- Get customer information.
+- Place an order by using a line of credit.
+  - Checks if the cost of the order is below the customer's credit limit.
+  - If the check passes, records the order history and updates the amount the customer has spent.
+- Get order information by order ID.
+- Get order information by customer ID.
+- Make a payment.
+  - Reduces the amount the customer has spent.
 
-### Transactions
+## Prerequisites
+
+- One of the following Java Development Kits (JDKs):
+  - [Oracle JDK 8](https://www.oracle.com/java/technologies/downloads/#java8)
+  - [Oracle JDK 11](https://www.oracle.com/java/technologies/downloads/#java11)
+  - [OpenJDK 8](https://openjdk.org/install/)
```

Comment:
Why only 8 for OpenJDK?

Should we say something like below as a note?
`Other JDKs should work, but we haven't tested with them.`

@brfrn169 Thought?

Reply:
@feeblefakie @josh-wong 
I agree. Given the many JDK distributions available, we should make a note of that.

Additionally, we need to make the same updates in the following section:
https://github.com/scalar-labs/scalardb/blob/e7ae27ea823555be73379af7984f78a6cf2afe9e/docs/getting-started-with-scalardb.md#install-a-jdk

Reply:
@feeblefakie @brfrn169 Understood. I've added a note about this to the **Prerequisites** section. 

PTAL at the attached screenshot for how it will look on the docs site. If this looks OK, I'll add it to other docs as well!

<img width="736" alt="screenshot" src="https://github.com/scalar-labs/scalardb-samples/assets/23216828/250e6771-d5f8-40f2-be3e-7e5038b252c5">

Reply:
@josh-wong Thank you for the update! I think we should also mention that ScalarDB supports Java 8 or later versions, and the JDKs provided in the list are just examples.

Reply:
So, something like this?

- One of the following Java Development Kits (JDKs):
  - Oracle JDK 8 or later
  - OpenJDK 8 or later

And the `Note` as you wrote.

Reply:
@feeblefakie @brfrn169 
As long as users can install any version after Oracle JDK 8 (for example, Oracle JDK 20) and OpenJDK 8 (for example, OpenJDK 20), I'm OK with this change. Can you please confirm if that's correct?

(I ask because I thought I overheard someone having an issue when using Oracle JDK 20...)

Reply:
@josh-wong (CC: @kota2and3kan) From ScalarDB's perspective, users can use any version after Oracle JDK 8. However, the sample application uses Gradle version 7.6, which does not support JDK 20 according to [Gradle's official documentation](https://docs.gradle.org/current/userguide/compatibility.html). 

I suggest we upgrade the Gradle version to 8.0 in a separate PR and state in our documentation that ScalarDB supports JDK 8 and later. What are your thoughts?

Reply:
@brfrn169 Thank you for the explanation! If we can upgrade the Gradle version to 8.0 so that it supports JDK 20, I'm OK with mentioning "or later" for each version.

I've updated the text to reflect that.

Reply:
@feeblefakie @brfrn169 According to our discussion today, I've updated the [**Prerequisites**](https://github.com/scalar-labs/scalardb-samples/pull/49/files#diff-f797bcc279a625246b6b47fc8eab1a2aac9b51029332790022a2a8306824d10cR34) section to mention specific JDK LTS versions and a note about how other non-LTS versions may work. 

PTAL!🙇🏻‍♂️

Reply:
@josh-wong LGTM! So it looks like we don't need to upgrade the Gradle version for now. Thank you!